<a href="https://colab.research.google.com/github/fralfaro/MAT281_2023/blob/main/docs/labs/lab_031.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


# MAT281 - Laboratorio N°03




## Problema 01


<img src="https://imagenes.universia.net/gc/net/images/practicas-empleo/p/pr/pro/profesiones-con-el-avance-de-la-tecnologia.jpg" width="480" height="360" align="center"/>


EL conjunto de datos se denomina `ocupation.csv`, el cual contiene información de distintos usuarios (edad ,sexo, profesión, etc.).

Lo primero es cargar el conjunto de datos y ver las primeras filas que lo componen:

In [2]:
import pandas as pd

In [3]:
# load data
url='https://drive.google.com/file/d/1SW1_vWBLd50COj4FbQdZenqAwA0nLDqC/view?usp=drive_link'
url='https://drive.google.com/uc?id=' + url.split('/')[-2]

df = pd.read_csv(url, sep="|" )
df.head()

user_id  age gender  occupation zip_code
0        1   24      M  technician    85711
1        2   53      F       other    94043
2        3   23      M      writer    32067
3        4   24      M  technician    43537
4        5   33      F       other    15213

El objetivo es tratar de obtener la mayor información posible de este conjunto de datos. Para cumplir este objetivo debe resolver las siguientes problemáticas:

1.- ¿Cuál es el número de observaciones en el conjunto de datos?

In [3]:
df.shape[0] #da la cantidad de filas sin contar el nombre de las columnas

943

Se tiene una muestra de 943 personas

2.- ¿Cuál es el número de columnas en el conjunto de datos?

In [4]:
df.shape[1] #da el numero de columnas sin contar el indice

5

Se tiene un total de 5 columnas

3.- Imprime el nombre de todas las columnas

In [5]:
df.columns #da una lista con el nombre de las columnas

Index(['user_id', 'age', 'gender', 'occupation', 'zip_code'], dtype='object')

4.- Imprima el índice del dataframe

In [6]:
df.index

RangeIndex(start=0, stop=943, step=1)

5.- ¿Cuál es el tipo de datos de cada columna?

In [7]:
df.dtypes

user_id        int64
age            int64
gender        object
occupation    object
zip_code      object
dtype: object

6.- Describir el conjunto de datos (**hint**: .describe())

In [8]:
df.describe(include='all') #da todos los datos estadistica de las columnas

user_id         age gender occupation zip_code
count   943.000000  943.000000    943        943      943
unique         NaN         NaN      2         21      795
top            NaN         NaN      M    student    55414
freq           NaN         NaN    670        196        9
mean    472.000000   34.051962    NaN        NaN      NaN
std     272.364951   12.192740    NaN        NaN      NaN
min       1.000000    7.000000    NaN        NaN      NaN
25%     236.500000   25.000000    NaN        NaN      NaN
50%     472.000000   31.000000    NaN        NaN      NaN
75%     707.500000   43.000000    NaN        NaN      NaN
max     943.000000   73.000000    NaN        NaN      NaN

7.- Imprimir solo la columna de **occupation**.

In [5]:
df['occupation']

0         technician
1              other
2             writer
3         technician
4              other
           ...      
938          student
939    administrator
940          student
941        librarian
942          student
Name: occupation, Length: 943, dtype: object

8.- ¿Cuántas ocupaciones diferentes hay en este conjunto de datos?

In [14]:
df['occupation'].unique() #muestra una lista con todos los elementos de ocupación sin repetir

array(['technician', 'other', 'writer', 'executive', 'administrator',
       'student', 'lawyer', 'educator', 'scientist', 'entertainment',
       'programmer', 'librarian', 'homemaker', 'artist', 'engineer',
       'marketing', 'none', 'healthcare', 'retired', 'salesman', 'doctor'],
      dtype=object)

9.- ¿Cuál es la ocupación más frecuente?

In [18]:
df['occupation'].value_counts().index[0]  
# se usa value_counts() para ordenar los elementos por su cantidad de veces que aparecen y index[0] recupera
# la ocupación con mas frecuencia

'student'

La ocupación mas común es student (estudiante)

10.- ¿Cuál es la edad media de los usuarios?

In [19]:
df['age'].mean() #.mean() calcula el promedio de la columna 'age'

34.05196182396607

La edad promedio es de 34 años

11.- ¿Cuál es la edad con menos ocurrencia?

In [21]:
df['age'].value_counts().tail()
# se usa value_counts() para ordenar los elementos por su cantidad de veces que aparecen y tail() muestra los ultimos elementos,
# i.e. lo menos recurrente

7     1
66    1
11    1
10    1
73    1
Name: age, dtype: int64

Las edades menos recurrentes son 7,66,11,10 y 73